src/main/webapp/file-download-popup-fragment.xhtml

```diff
@@ -121,56 +123,83 @@
-                           update="guestbookUIFragment">
-                <f:param name="DO_GB_VALIDATION" value="true"/>
+            <p:commandButton styleClass="btn btn-default" value="#{bundle.acceptTerms}"
+                             rendered="#{guestbookResponse.fileFormat != 'externalTool' and
+                                         guestbookResponse.fileFormat != 'worldMap' and
+                                         guestbookResponse.fileFormat != 'package' and
+                                         popupContext != 'previewTab'}"
+                             actionListener="#{fileDownloadHelper.writeGuestbookAndStartDownload(guestbookResponse)}"
+                             update="guestbookUIFragment">
+                <f:param name="DO_GB_VALIDATION_#{popupContext}" value="true"/>
+            </p:commandButton>
+            <!--PREVIEW TAB BUTTON-->
+            <p:commandButton styleClass="btn btn-default" value="#{bundle.acceptTerms}"
+                             rendered="#{popupContext == 'previewTab'}"
+                             actionListener="#{FilePage.showPreview(guestbookResponse)}"
+                             process="previewTab"
+                             update="fileForm:tabView">
+                <f:param name="DO_GB_VALIDATION_#{popupContext}" value="true"/>
             </p:commandButton>
+            <!--WORLDMAP BUTTON-->
             <p:commandButton styleClass="btn btn-default" value="#{bundle.acceptTerms}" rendered="#{guestbookResponse.fileFormat == 'worldMap'}"
-                            actionListener="#{fileDownloadHelper.startWorldMapDownloadLink(guestbookResponse, null)}" 
-                            update="guestbookUIFragment">
-                <f:param name="DO_GB_VALIDATION" value="true"/> 
+                             actionListener="#{fileDownloadHelper.startWorldMapDownloadLink(guestbookResponse, null)}"
+                             update="guestbookUIFragment">
+                <f:param name="DO_GB_VALIDATION_#{popupContext}" value="true"/>
             </p:commandButton>
+            <!--EXTERNAL TOOL BUTTON-->
             <!--On the dataset page (but not the file page), "tool" is null so we get the tool from the guestbookResponse.-->
-            <p:commandButton styleClass="btn btn-default" process="@this" value="#{bundle.acceptTerms}" rendered="#{guestbookResponse.fileFormat == 'externalTool'}"
-                           action="#{fileDownloadHelper.writeGuestbookAndLaunchExploreTool(guestbookResponse, fileMetadata, tool)}"
-                           update="guestbookUIFragment">
-                <f:param name="DO_GB_VALIDATION" value="true"/>
+            <p:commandButton styleClass="btn btn-default" value="#{bundle.acceptTerms}" rendered="#{guestbookResponse.fileFormat == 'externalTool'}"
+                             action="#{fileDownloadHelper.writeGuestbookAndLaunchExploreTool(guestbookResponse, fileMetadata, tool)}"
```

Comment:
If we are rendering the buttons based on the gbResponse file formats, can we just have one method for Launching the explore tool, world map and package popup that decides what to do based on the file format. That could reduce three buttons to one.

Reply:
A related thought: @djbrooke  has given the ok to start removing world map related buttons and such, so if that helps simplify the code... (I've started to do that in the kebab issue / edit buttons)

Reply:
Yes - I have on my list to start clearing out Worldmap issues from Github and to create an issue to pull out any Worldmap/Geoconnect code specific in Dataverse, so it's good to get a headstart.

Reply:
I just looked at this and I don't see a lot of value in refactoring. The complexity has to live somewhere and right now a decision is made on the front end. I think it's fine as it is.